docs: Add information regarding ESTALE error in troubleshooting guide (#3654)
GoogleCloudPlatform/gcsfuse: docs/troubleshooting.md

It’s a generic error, but the most probable culprit is the bucket not having the right permission for Cloud Storage FUSE to operate on. Ref - [here](https://stackoverflow.com/questions/36382704/gcsfuse-input-output-error)

### Stale File Handle Error (ESTALE)

> This error occurs when GCSFuse detects that a file has been modified or deleted on GCS by another process or mount since it was opened. This is a data integrity feature that prioritizes providing clear indications of potential conflicts, preventing silent data loss.

Reviewer: ```suggestion
This error indicates that the file being accessed has been modified or deleted on GCS externally since the GCSFuse mount opened it. This is a data integrity feature that prioritizes providing clear indications of potential conflicts, preventing silent data loss.
```
